examples/src/main/java/com/mapbox/navigation/examples/core/InstructionViewActivity.kt

```diff
@@ -0,0 +1,429 @@
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        mapView.onPause()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        mapView.onStart()
+
+        mapboxNavigation.registerLocationObserver(locationObserver)
+        mapboxNavigation.registerTripSessionStateObserver(tripSessionStateObserver)
+    }
+
+    override fun onStop() {
+        super.onStop()
+        mapView.onStop()
+
+        mapboxNavigation.unregisterLocationObserver(locationObserver)
+        mapboxNavigation.unregisterTripSessionStateObserver(tripSessionStateObserver)
+        mapboxNavigation.unregisterRouteProgressObserver(routeProgressObserver)
+        mapboxNavigation.unregisterBannerInstructionsObserver(bannerInstructionObserver)
+        mapboxNavigation.unregisterVoiceInstructionsObserver(voiceInstructionsObserver)
+
+        stopLocationUpdates()
+    }
+
+    override fun onLowMemory() {
+        super.onLowMemory()
+        mapView.onLowMemory()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        mapView.onDestroy()
+
+        mapboxNavigation.stopTripSession()
+        mapboxNavigation.onDestroy()
+
+        speechPlayer.onDestroy()
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        mapView.onSaveInstanceState(outState)
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        Timber.d("onMapReady")
+        this.mapboxMap = mapboxMap
+        mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+
+        mapboxMap.addOnMapLongClickListener { latLng ->
+            Timber.d("onMapLongClickListener position=%s", latLng)
+            destination = latLng
+            locationComponent?.lastKnownLocation?.let { originLocation ->
+                mapboxNavigation.requestRoutes(
+                    RouteOptions.builder().applyDefaultParams()
+                        .accessToken(Utils.getMapboxAccessToken(applicationContext))
+                        .coordinates(originLocation.toPoint(), null, latLng.toPoint())
+                        .alternatives(true)
+                        .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                        .build(),
+                    routesReqCallback
+                )
+            }
+            true
+        }
+
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            locationComponent = mapboxMap.locationComponent.apply {
+                activateLocationComponent(
+                    LocationComponentActivationOptions.builder(this@InstructionViewActivity, style)
+                        .build()
+                )
+                cameraMode = CameraMode.TRACKING
+                isLocationComponentEnabled = true
+            }
+
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap).also {
+                it.addProgressChangeListener(mapboxNavigation)
+                it.setCamera(DynamicCamera(mapboxMap))
```

Comment:
Why do we need these 2 lines ☝️ ?

Reply:
I don't know the exact reason, but I referred here: https://github.com/mapbox/mapbox-navigation-android/blob/master/examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt#L185-L186

@cafesilencio might knows about it.

Reply:
Should we remove it and see if things work the same?

Reply:
so Ii think it's required by using camera, otherwise it will crash when you `startCamera` since there is no camera has been set.